File: CS/WindowsApplication1/XtraReport1.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using DevExpress.XtraReports.UI;
using System.Data;

namespace WindowsApplication1 {
    public partial class XtraReport1: DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }
        Hashtable ht = new Hashtable();
        private void GroupCategoryCounter_GetValue(object sender,GetValueEventArgs e) {
            string cat = ((DataRowView)e.Row)["CategoryName"].ToString();
            int count = 0;
            if(!ht.ContainsKey(cat)) {
                count = nwindDataSet1.CategoryProducts.Select("CategoryName = '" + cat + "'").Length;
                ht.Add(cat,count);
            } else 
                count = Convert.ToInt32(ht[cat]);
            
            e.Value = count;
        }

    }
}

File: VB/WindowsApplication1/XtraReport1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraReports.UI
Imports System.Data

Namespace WindowsApplication1
	Partial Public Class XtraReport1
		Inherits DevExpress.XtraReports.UI.XtraReport
		Public Sub New()
			InitializeComponent()
		End Sub
		Private ht As New Hashtable()
		Private Sub GroupCategoryCounter_GetValue(ByVal sender As Object, ByVal e As GetValueEventArgs) Handles GroupCategoryCounter.GetValue
			Dim cat As String = (CType(e.Row, DataRowView))("CategoryName").ToString()
			Dim count As Integer = 0
			If (Not ht.ContainsKey(cat)) Then
				count = nwindDataSet1.CategoryProducts.Select("CategoryName = '" & cat & "'").Length
				ht.Add(cat,count)
			Else
				count = Convert.ToInt32(ht(cat))
			End If

			e.Value = count
		End Sub

	End Class
End Namespace
